app/controllers/api/v3/courses_controller.rb

```diff
@@ -6,7 +6,11 @@ class CoursesController < API::V3::ApplicationController
       before_action :build_courses
 
       def index
-        course_search = CourseSearchService.call(filter: params[:filter], sort: params[:sort], course_scope: @courses)
+        course_search = CourseSearchService.call(
+          filter: params[:filter],
+          sort: params[:sort],
+          course_scope: @courses.published,
```

Comment:
Probably want to move that in to `CourseSearchService` defaulting with published `true`

Reply:
@defong & @gpeng encapsulating that extra condition in `CourseSearchService` would be neater but it looks like it would change behaviour, e.g. these two endpoints could get out of step https://github.com/DFE-Digital/teacher-training-api/blob/master/app/controllers/api/public/v1/providers/courses_controller.rb#L6-L16

I'll try to work out how many courses this issue applies to...

Reply:
It looks like there are around 246 courses in the current cycle that only have a withdrawn CourseEnrichment. I don't know how many of those are searchable.

Reply:
https://api.publish-teacher-training-courses.service.gov.uk/api-reference.html#schema-coursefilter

That should of covered us, as it purely a bug.

![image](https://user-images.githubusercontent.com/470137/123605972-500db600-d7f4-11eb-979b-e20c48939fa8.png)

See this just suggest to me, that we haven't understood the root cause for the bug, as `findable` was a long establish meaning of being on `find` .
https://github.com/DFE-Digital/teacher-training-api/blob/5f2bbefe7a76c35b007396d0b86720811251a7a3/app/models/course.rb#L526-L531

Reply:
@defong I'm a bit confused I must admit. Having `findable` as the single determining factor would be nice but it seems more complicated than that. 

https://github.com/DFE-Digital/teacher-training-api/blob/master/app/controllers/api/v3/courses_controller.rb#L21 is filtering out courses unrelated to `findable`. Trouble is that check is only done on the `show` action, not on the `index`.

Reply:
@defong So do you think this is a data problem rather than something needing a code change?

Reply:
I think we need to understand how many course this has this issues, it maybe an more feasible to fix as a data issue.

Otherwise we do the code fix and follow it through in order to deem the accuracy of https://api.publish-teacher-training-courses.service.gov.uk/reporting

Reply:
@defong it looks like there are only two such courses in the current cycle.

Thats based on some rough-and-ready queries I just did: Provider 5A1 course G011 and provider 1EF course A431. (I ran a query that collects all the 'withdrawn' courses, defined as those that just have a withdrawn enrichment, and intersecting them with the results of the `findable` scope).

What do you think is the right thing to do here?

Reply:
Yeah I think fixing the underling data should be the path of least resistance.